### PR TITLE
[eas-cli] Update: improve DX on first run, correct error message

### PR DESCRIPTION
**Pull request by Simek**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Fixes ENG-6013
Fixes ENG-6014

# How

This changeset adds the prompt for the user when there is no runtime version specified, which asks if setup the updates automatically for them (via `eas update:configure`).

If user do not want this the CLI exits as before.

If user selects "Yes", we are programmatically running `eas update:configure`, and then re-reading the modified config, so the process can continue without the interruption or command exit.

As always, the copy/messages content probably can be written better. 😉 

# Test Plan

The changes have been tested locally in few project by running `dev` bin of EAS CLI.

# Preview

Screenshot 2022-10-07 at 21 43 45

Screenshot 2022-10-07 at 21 40 31

### CODE REVIEW

**Comment by Simek** [comment]
/changelog-entry new-feature Update: Ask user to run `eas update:configure` automatically, if `runtimeVersion` is missing.
/changelog-entry chore Update: Improve the error message if `runtimeVersion` is missing.

File: packages/eas-cli/src/commands/update/index.ts
Comment (kbrandwijk) [inline packages/eas-cli/src/commands/update/index.ts:619]:
```suggestion
        title: 'No, I will set the runtime version manually (EAS CLI exits)',
```

Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:619]: How strong is you opinion on that phrasing? 

The `eas update:configure` besides setting the runtime version performs edit of other files, to prepare the project for EAS Update, that's why I have decided to went with that sentence.

Comment (kbrandwijk) [reply packages/eas-cli/src/commands/update/index.ts:619]: The 'I will edit files' part is a bit inconclusive, and you are only checking for runtime version, so it's not guaranteed that they need to run that command, they could have made any change that causes this. So if you're checking runtime version, I think the instructions should be as narrow as possible to set runtime version. Otherwise, instead of saying 'cannot detect runtime version' you should say 'it seems your project is not configured for EAS Update yet' (but I prefer to just stick to the runtime version.

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/index.ts:628]: this is the first time we've done this in the CLI so it may be setting a bit of a precedent. I see that https://oclif.io/docs/running_programmatically says it's doable, but it also mentions that factoring it out is preferred and I tend to agree. The main reason I think we should factor out the underlying code is that that there's no typechecking on this passing of flags, so if we add a flag to the configure command, we'll most likely forget to inspect this callsite to ensure the flag is passed. By factoring out the code, we can type the method strictly.

**Review by wschurman — CHANGES_REQUESTED** [review]
Couple issues I see:
1. We'll need to handle the non-interactive flag here (fail same as before if the flag is set).
2. The configs fetched on L203 and L207 are now stale, so they'll need to be re-fetched.
3. The installation of expo-updates in L215-L239 now I believe can be removed since it's done in the call of configure?

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:628]: According to dynamic imports, Expo CLI use this pattern for the performance, I also saw one usage in this codebase already for loading config file:
* https://github.com/expo/eas-cli/blob/024c699637f46914d0ba9766f321faaf5fc4efb4/packages/eas-cli/src/metadata/config.ts#L23

Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:628]: Ah, sorry, I didn't mean the dynamic import, I meant the calling of another command programmatically instead of just factoring out the code. Dynamic imports are totally cool.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:624]: What do you think about: 
```
"No, I will set the runtime version or run the command manually (EAS CLI exits)"
```
since the question above refers only to that action?

And then, we can rephrase the question above: 
```
"Running 'eas update:configure' will attempt to fix this issue. Do you want us to run the command for you?"
```
Or:
```
"Running 'eas update:configure' will try to automatically add the missing configuration. Do you want us to run the command for you?"
```
Or even something else, along those lines.

**Comment by Simek** [comment]
1. Nice catch, addressed.
2. Public config is now passed from the transform method, and updated if needed. There is no need to refetch the private config, since it's only used once, before the transform method call.
3. I was still able to trigger this warning when using `--non-interactive` flag in the first version of this PR, but now, after refactor, we only call the `eas.json` edit part, so this is definitely still needed.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts]: After the latest changes the content of this message has been altered, since I have changed the logic. Let's resolve that for now, and we can get back to this in the re-review, if I get confirmation, that only calling the `app.json` edit was a correct approach here.

File: packages/eas-cli/src/project/workflow.ts
Comment (wkozyra95) [inline packages/eas-cli/src/project/workflow.ts:40]: maybe inline that type in the function return value or name it sth more descriptive e.g. `WorkflowPerPlatform`, plus you probably should use `Record<Platform, Workflow>` instead

File: packages/eas-cli/src/project/workflow.ts
Comment (dsokal) [inline packages/eas-cli/src/project/workflow.ts:44]:
```suggestion
): Promise<Record<Platform, Wokrflow>> {
```

File: packages/eas-cli/src/project/workflow.ts
Comment (dsokal) [inline packages/eas-cli/src/project/workflow.ts:40]:
```suggestion
```

File: packages/eas-cli/src/project/workflow.ts
Comment (dsokal) [inline packages/eas-cli/src/project/workflow.ts:48]:
```suggestion
  const [android, ios] = await Promise.all([
    resolveWorkflowAsync(projectDir, Platform.ANDROID),
    resolveWorkflowAsync(projectDir, Platform.IOS),
  ]);
  return { android, ios };
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [inline packages/eas-cli/src/commands/update/index.ts:623]: What's that? Do you want to print the `x` sign? You can do it explicitly. See how we implemented `Log.succeed` - https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/log.ts#L48

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [inline packages/eas-cli/src/commands/update/index.ts:202]: Don't use `let`

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [inline packages/eas-cli/src/commands/update/index.ts:642]: Why do you need the dynamic import?

File: packages/eas-cli/src/commands/update/index.ts
Comment (wkozyra95) [inline packages/eas-cli/src/commands/update/index.ts:620]: You are still in a file where a command is defined, so it might be ok, but in general, you should avoid exiting anywhere other than top-level code and use exceptions instead.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:642]: Performance, it's the common pattern in Expo CLI.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:202]: I also prefer `const`, but Will requested that change:
> The configs fetched on L203 and L207 are now stale, so they'll need to be re-fetched.

Using `let` here is justified I think.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:623]: There was not a helper for that, so I have used Ora directly.

I always attempt to reduce the amount of lines changed, if not needed, but probably we can add that helper, since this PR evolved fince it's first form already.

File: packages/eas-cli/src/project/workflow.ts
Comment (Simek) [reply packages/eas-cli/src/project/workflow.ts:48]: Why?

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:642]: It's not Expo CLI, please don't use dynamic import in this repo.

File: packages/eas-cli/src/project/workflow.ts
Comment (dsokal) [reply packages/eas-cli/src/project/workflow.ts:48]: Performance :D

Comment (Simek) [reply packages/eas-cli/src/project/workflow.ts:44]: Can you alaborate?

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:642]: Also, this is "need" IHMO, since this path will be triggered only on non-setup or invalid setup projects which make the percentile of `update` calls, and with static import we will always load whole configure step, which affects the command stat-up and TTI time.

Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:642]: Can you prepare some benchmarks on what is the performance gain? I'd say it is negligible.

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:623]: Let's change that

File: packages/eas-cli/src/project/workflow.ts
Comment (dsokal) [reply packages/eas-cli/src/project/workflow.ts:44]: Ok, I'll ask another question. You changed it from `Promise.all` to the non-`Promise.all` version. Why?

Comment (dsokal) [reply packages/eas-cli/src/project/workflow.ts:44]: With `Promise.all` those two can run concurrently.

Comment (wkozyra95) [reply packages/eas-cli/src/project/workflow.ts:44]: Your implementation is running those methods one after another, Dominik's approach is running them at the same time.

File: packages/eas-cli/src/commands/update/index.ts
Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:642]: I'm not saying about benchmarks in general. I'm saying that I want to see benchmarks for this particular case. As Wojtek pointed out, there is no real gain here as probably all other modules imported in the `configure` module are imported anyway. 
The problem with the code that you proposed is that using dynamic imports in **EAS CLI** would be a precedence. In this case, I'm afraid, it's not at all justified.

Comment (wkozyra95) [reply packages/eas-cli/src/commands/update/index.ts:639]: Also unless ocliff is doing lazy loading per command there might be no impact on performance at all regardless of the size of the imported part.

Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:639]: https://github.com/expo/eas-cli/pull/1427#discussion_r991500606

https://github.com/expo/eas-cli/blob/024c699637f46914d0ba9766f321faaf5fc4efb4/packages/eas-cli/src/metadata/config.ts#L23

Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:639]: This line is reading the config from the user's project. It's not the same. 
In you're code you're importing internal modules.

Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:639]: > there is no real gain here as probably all other modules imported in the configure module are imported anyway.

There are no other `configure` imports or usage in the `update` command, and imports are loaded per command on run, so I think that the sentence above is incorrect.

Comment (dsokal) [reply packages/eas-cli/src/commands/update/index.ts:639]: Just compare the imports in those files.
Screenshot 2022-10-12 at 12 53 31
Screenshot 2022-10-12 at 12 53 17

I won't believe there's any performance gain until I see data :)

Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:639]: Now that this is just a normal function being imported I think it it makes sense to just use a normal import (we do this for pretty much all business logic functions). 

The key though is that we should move the `configureAppJSONForEASUpdateAsync` out of the command module and into its own module. If it's in the command module, a lot of extra stuff pertaining to commands and command static stuff is loaded and I'd be willing to bet that the dynamic import is more optimal since it wouldn't be loaded in most cases. If it's in its own module, then none of the command-related stuff will be loaded and the overhead of loading it normally is negligible. If it's in its own module, it is pretty much the same as any other function being imported in any command module.

File: packages/eas-cli/src/log.ts
Comment (wschurman) [inline packages/eas-cli/src/log.ts:48]:
```suggestion
  public static fail(message: string): void {
```
(to match tense of `succeed`. I'd also be open to renaming `succeed` to `success` and make this one `failure`)

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:617]: Yep, the common pattern for this is to just throw a normal exception. (grep for `if (nonInteractive)`).

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/index.ts:636]: ditto about throwing an exception rather than using `Errors.exit` (can grep for both to see the relative usage of each, and tend to prefer the more common pattern)

**Review by wschurman — CHANGES_REQUESTED** [review]
Looking a lot better and more well-typed (thanks for making that change)! 

The main change requested in this round of reviews is to move the factored-out-functions into a module outside of the command (which should alleviate concerns about using normal imports over dynamic imports).

I don't really have a good sense of dynamic import speed, but generally I opt for the more common pattern in a codebase when adding new code. I suggest sticking to the more common convention for feature PRs in order to reduce unrelated discussion on the PR and to get your feature landed more quickly. It was my bad for my comment conveying that it should be used; instead, I was more was conveying that that wasn't the issue with the line, but instead the issue was the import of the whole command (which is now fixed).

That being said, if there's a convention that you believe is better (and in this case have some perf numbers), I definitely encourage you to put up some PRs to change the pattern throughout the codebase. I tend to do this often. I often start with a small proposal PR showing the idea, gather comments, and then a few larger PRs to change the convention once there's buy in.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:651]: This is a conscious decision to exit without an error here, since the user have chose this option. Throwing an another error here will not add any value or information, can only confuse user.

https://github.com/expo/eas-cli/blob/123615212058bf841dec00edcc16da284c27e148/packages/eas-cli/src/commands/build/index.ts#L300-L307

Comment (wkozyra95) [reply packages/eas-cli/src/commands/update/index.ts:651]: > This is a conscious decision to exit without an error here,

`Errors.exit(1)` is not exiting without an error.

>  can only confuse user.

I don't see how, if you throw sth along the lines of "Aborting ..."

> Also this pattern has been taken from the codebase...

Similar patterns are not really a good argument, sth that is fine there might not be fine here. This "pattern" is used in very different contexts:
- In build, it's used in code that is run at the very start of the process, as part of the input/config validation. That function is handling deprecated config for something very specific.
- In update code that exit is run in the util function that someone might want to extract to common code in the future.

With that said I think leaving exit might be fine, but in this case, I think it's a bit borderline.

Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:651]: Under the hood all `Errors.exit` does is throw an error: https://github.com/oclif/core/blob/main/src/errors/index.ts#L18

I don't feel too strongly here, but now that I grep for similar cases I see a mix of `throw`, `Errors.exit` and `process.exit`. `process.exit` is definitely incorrect since it will cause the tear down code of oclif commands not to run (see EasCommand.finally). `Errors.exit` and `throw` are equivalent, so it's just a matter of style.

I'd say this is fine here, but really what we should do in a follow-up is a brief audit of all cases followed by a potential lint rule to ensure that this remains consistent going forward across the codebase. That way there's no attention cost to this anymore.

For this PR, it's fine to keep it as you have it.

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/index.ts:252]: Prefer to keep this named `exp` and renaming the one before it `expBeforeRuntimeVersionUpdate` (or using `let` is fine as well). The reason for this is that L429 is missing a change to `expUpdated`.

**Review by wschurman — APPROVED** [review]

**Review by jonsamp — COMMENTED** [review]
Screen Shot 2022-10-17 at 1 32 13 PM

The decisions + code in this PR are right to me. I tested this locally and see that I run into an error (pictured above): `Error: An SDK version must be defined when using the 'sdkVersion' runtime policy.`. --However, when I run the same `eas update` command again, it works fine.

Once we get this to publish with the written runtime version, we should be good to ship this.

**Comment by Simek** [comment]
@jonsamp Last commit added after your comment should fix those issues, as we discussed, it would be great if you can confirm that the problem with runtime policy no longer occurs.

**Review by wschurman — CHANGES_REQUESTED** [review]
https://github.com/expo/eas-cli/pull/1467 brings up a really good point that I hadn't thought of: installing or configuring expo-updates (or changing existing configuration) requires a new build. Because of this, doing these things at the beginning of `eas update` (including installing expo-updates) might not make as much sense as I once thought, since even if the update is published the user would still need to do a build which would embed a newer manifest in it and the published update would never be used (the clients select the most recent update, and in this case the build's embedded update would be newer than this published update). To me, this behavior seems more confusing than just throwing an error and telling people to run `eas update:configure`, which with https://github.com/expo/eas-cli/pull/1467 will tell the user they need to run a build before published updates will work.

**Comment by jonsamp** [comment]
We talked about this feature in a meeting for awhile. @brentvatne put together a summary:

> should eas update run eas update:configure automatically if not yet configured? imo yes -> but let people know afterwards that the update may not apply to any existing builds.

> should eas update and eas build run eas init automatically if not yet configured? imo yes -> similar to trying to access an authenticated portion of a sign before signing in. i think it's a bad first time experience for a user to run a command and have it fail and tell them to run another command manually

> should we configure eas update and build in eas init? yes, we should prompt people for this

> should we configure eas update when you run eas build? yes, we should use the same sort of prompt we use around push notifications to let people say yes/no/don't ask me again

I think that we should prompt people on EAS Update and as the first one says "let people know afterwards that the update may not apply to any existing builds". Developers can still view published updates in Expo Go and development builds even if they did not have EAS Update configured until they published.

@wschurman, the underlying premise of this PR is right to me. If you agree, and you think the code is good, we should approve this one.

**Review by jonsamp — APPROVED** [review]
I just tested this and it is working great. Nice work Simek, this is a big improvement!

**Review by brentvatne — APPROVED** [review]

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [inline packages/eas-cli/src/commands/update/index.ts:643]:
```suggestion
      `Configure runtime version in ${chalk.bold(
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [inline packages/eas-cli/src/commands/update/index.ts:645]:
```suggestion
      )} automatically for EAS Update?`,
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [inline packages/eas-cli/src/commands/update/index.ts:680]:
```suggestion
      `Continue update process with uncommitted changes in repository?`,
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [inline packages/eas-cli/src/commands/update/index.ts:684]:
```suggestion
          title: 'No, I will commit the modified files first',
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [inline packages/eas-cli/src/commands/update/index.ts:649]:
```suggestion
          title: 'No, I will set the runtime version manually',
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/index.ts:677]: Based on the code in `configure`, I think the result of `configureAppJSONForEASUpdateAsync` is the source of truth for `newConfig`.

```suggestion
    const newConfig = await configureAppJSONForEASUpdateAsync({
      exp,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
    });

    await configureNativeFilesForEASUpdateAsync({
      exp: newConfig,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
      graphqlClient,
    });
```

File: packages/eas-cli/src/commands/update/configure.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/configure.ts:58]: optional: can put all three of these methods into a single utility method to ensure they are kept consistent between the two commands
```typescript
function configureAsync(...) {
  const workflows = await resolveWorkflowPerPlatformAsync(projectDir);
  const updatedExp = await configureAppJSONForEASUpdateAsync(...);
  await configureNativeFilesForEASUpdateAsync(...)
}
```
and then call this function from both `configure` and `update`.

**Review by wschurman — APPROVED** [review]
We'll want to ensure @FiberJW's changes are included in the common factored out code so that this happens on both `configure` and `update`
> let people know afterwards that the update may not apply to any existing builds

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:677]: @wschurman I have stated the reasoning for how this code works in conversation with Jon (should post a comment here probably), let me quote it:

> the problem was that `configureAppJSONForEASUpdateAsync` returns updated config, but it was only a partial with the changes, so instead of using result directly we need to merge it with full config

It was a root cause why `"Error: Unable to determine runtime version for android"` started appearing after one of the PR changes. The problem has been mentioned in this comment:
* https://github.com/expo/eas-cli/pull/1427#pullrequestreview-1144562679

Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:677]: Looking at `modifyConfigAsync` source, it looks like it returns exactly what's on disk? Maybe we need to run `transformRuntimeVersions` again or something?

Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:677]: I don't think that `modifyConfigAsync` is a source of problem there, it already receives config partial:

* https://github.com/expo/eas-cli/blob/b9c505bbdc30af90b17633028c171465e9ec8df2/packages/eas-cli/src/commands/update/configure.ts#L128-L129

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:684]: This pattern of adding `(EAS CLI exits)` at the end of exiting option has been taken from the existing code.

I we want to remove it, I feel like this should be done in the whole codebase, in the separate PR, but personally I like this addition, and I think that make it clear for the user what will happed.

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [reply packages/eas-cli/src/commands/update/index.ts:675]: Ah, I see. In that case we should re-fetch the full config using `getDynamicProjectConfigAsync`. Doing just an object merge will overwrite keys (doesn't do a deep merge).

```suggestion
    const configUpdate = await configureAppJSONForEASUpdateAsync({
      exp,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
    });

    const newConfig = (await getDynamicProjectConfigAsync({ isPublicConfig: true })).exp;

    await configureNativeFilesForEASUpdateAsync({
      exp: newConfig,
      projectDir,
      projectId,
      platform: platformFlag as RequestedPlatform,
      workflows,
      graphqlClient,
    });
```

File: packages/eas-cli/src/commands/update/index.ts
Comment (wschurman) [inline packages/eas-cli/src/commands/update/index.ts:638]: catching all errors assumes that they are all runtime version errors vs other classes of errors (code errors, other errors). There doesn't seem to be a very good way of detecting just the class of error though other than error message introspection (which I don't recommend), so I think this is okay.

The only thing we risk is that it is already configured and `transformRuntimeVersions` throws another error and we clobber their configuration accidentally.

**Review by wschurman — APPROVED** [review]
Approving with the a few inline comments (just saw you already addressed one of them and the other isn't actionable).

File: packages/eas-cli/src/commands/update/configure.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/configure.ts:58]: I think that @byCedric plan to refactor the `configure` code anyway, and if this is out of scope of his planned changes I can tackle that in a separate PR.

File: packages/eas-cli/src/commands/update/index.ts
Comment (Simek) [reply packages/eas-cli/src/commands/update/index.ts:645]: Only error that `transformRuntimeVersions` can throw is a one defined by us via `nullthrows` (it's a very basic method), so I think that is safe to use the generic catch here, and assume that every error is a missing runtime version.

File: packages/eas-cli/src/commands/update/index.ts
Comment (brentvatne) [reply packages/eas-cli/src/commands/update/index.ts:693]: It looks like there is only one other place where we do this and it's in a fairly obscure place: [https://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305](https://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305)



I'm not entirely opposed to letting users know that the process will exit but I found this wording to be confusinghttps://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305. It'd be better to say something like (this process will exit)

File: packages/eas-cli/src/commands/update/index.ts
Comment (wkozyra95) [reply packages/eas-cli/src/commands/update/index.ts]: Sorry, it appears I was wrong (I relied on someone else comment without verifying that myself), It's fine to use `Errors.error` or `Errors.exit` anywhere in the code, it throws an exception not exiting immediately as I thought before.

**Comment by jonsamp** [comment]
I would like to get this merged whenever you have a chance @Simek. This PR is in a good spot and I'd love to see it in production sooner than later. I really appreciate your work on this and your responses to all the comments.

**Pull request merged**
